### DISCUSSION TITLE
🚀 Feature Request:类似@tanstack/vue-query 注册全局错误类型的功能

### DISCUSSION THREAD

**Discussion by songpeng154** [alovajs/alova#614] [Feature Request]
### What problem does the new feature need to solve?

统一的错误处理
[https://tanstack.com/query/latest/docs/framework/vue/typescript#typesafe-disabling-of-queries-using-skiptoken](vue-query)
```ts
import type { ResponseError } from '@/services/request/axios/types'
import '@tanstack/vue-query'

declare module '@tanstack/vue-query' {
  // 全局注册
  interface Register {
    // 注册全局错误
    defaultError: ResponseError
  }
}

```

### How should the new features you expect looks like?
```ts
import type { ResponseError } from '@/services/request/axios/types'
import 'alova'

declare module 'alova' {
  // 全局注册
  interface Register {
    // 注册全局错误
    defaultError: ResponseError
  }
}

```
_No response_

### How important is it to you?

I'll give up alova without it

**Comment by MeetinaXD** [comment]
感谢提出建议！这个 feature 应该有不少人会需要，但有一些问题：

1. alova 的类型依附于实例和 Method，需要其他方式定义错误类型；
2. 捕获异常时，error的实际类型可能与声明不一致。在这种前提下，声明类型的效果不如 type guard

这两点是我们正在考虑的，如果你有更好的的想法，也可以告诉我们。
